### ISSUE TITLE
DSC v3 supports using class-based resources cross platform

### ISSUE THREAD

**Issue by sdwheeler** [MicrosoftDocs/PowerShell-Docs#7373]
# Summary of the new document or enhancement

- List of articles that need to be updated:
- New articles that need to be created:
- Link(s) to related code PR(s) in the PowerShell/PowerShell repo: https://github.com/PowerShell/PowerShell/pull/13399
- Link(s) to related issue(s) in the PowerShell/PowerShell repo:

## Description of what changed

Together with PSDesiredStateConfiguration v3 module allows Get-DscResource, Invoke-DscResource and DSC configuration compilation on all platforms, supported by PowerShell (using class-based DSC resources).

At this point this PR is not a breaking change: existing (MOF/MMI-based) DSC code will continue to work as it was and will continue to be the default mechanism for DSC operations (for now).
Cross-platform DSC updates of this PR are done as an alternative mechanism for DSC operations; to use them:

- PSDesiredStateConfiguration v3 module has to be installed;
- PS7DscSupport experimental feature has to be enabled.

**Comment by chasewilson** [comment]
Putting this on hold until PSDesiredStateConfiguration V3]() is published. 

Since this feature is useless without it and PSDSC V3 isn't planned for preview until later, it's suggested to wait on the documentation. 

Main Findings to remember when documentation is completed:
- V3 doesn't support script resources period.
- `configuration` keyword is not planned to be supported at release
- Only `Invoke-DscResource` will be supported at release.
